benchmark/comprehensive_benchmark.dart

```diff
@@ -94,165 +94,156 @@ class _Runtime {
 
   _Operation _randomOperation() {
     final type = _OperationType.values[r.nextInt(_OperationType.values.length)];
+    const noop = (_OperationType.noop, 0);
     switch (type) {
+      case _OperationType.noop:
+        return noop;
       case _OperationType.add:
-        return _AddOperation(_randomElement());
+        return (type, _randomElement());
       case _OperationType.removeIdx:
         if (_set.isEmpty) {
-          return _AddOperation(_randomElement());
+          return noop;
         }
-        return _RemoveIdxOperation(r.nextInt(_set.length));
+        return (type, r.nextInt(_set.length));
       case _OperationType.removeElement:
         if (_set.isEmpty) {
-          return _AddOperation(_randomElement());
+          return noop;
         }
-        return _RemoveElementOperation(_set.elementAt(r.nextInt(_set.length)));
+        return (type, _set.elementAt(r.nextInt(_set.length)));
       case _OperationType.removeWhere:
-        return _RemoveWhereOperation(_randomElement());
+        return (type, _randomElement());
       case _OperationType.visit:
-        return _VisitOperation(_randomElement());
+        return (type, _randomElement());
       case _OperationType.iterateThenAdd:
-        return _IterateThenAddOperation(_randomElement());
+        return (type, _randomElement());
       case _OperationType.iterateThenRemove:
-        return _IterateThenRemoveOperation(_randomElement());
+        return (type, _randomElement());
     }
   }
 
   int _randomElement() => r.nextInt(_maxElement) + 1;
 }
 
 enum _OperationType {
+  noop(_noopOperation),
```

Comment:
sadly I can't seem to override a method per instance of enum in dart

I could otherwise have just a giant switch but I thought this was slightly more organized

Reply:
Just have a static method as a proxy?

Reply:
isn't that exactly what I have rn?

Reply:
Haha sorry, I wrote that review from the bar's toilet. 😂 What about using a sealed class instead and just switch over that?

Reply:
I replaced the enum with a custom selaed class (basically make your own enum bc Dart's don't support custom overloads)